src/target/source/codegen_cuda.cc

```diff
@@ -274,9 +274,21 @@ void CodeGenCUDA::PrintVecElemLoad(
   static const char access[] = {'x', 'y', 'z', 'w'};
   CHECK(i >= 0 && i < (t.is_float16() ? 8 : 4));
   if ((t.is_int()) && t.bits() == 8) {
-    os << "((char)(" << vec << " >> " << i * 8 << "))";
+    if (t.lanes() == 1) {
+      os << vec;
+    } else if (t.lanes() == 2) {
+      os << vec << "." << access[i % 2];
+    } else {
+      os << "((char)(" << vec << " >> " << i * 8 << "))";
+    }
   } else if ((t.is_uint()) && t.bits() == 8) {
-    os << "((unsigned char)(" << vec << " >> " << i * 8 << "))";
+    if (t.lanes() == 1) {
```

Comment:
This "lane = 1" logic could be promoted for all cases.

Reply:
When the lane is 1, the type is "char",  it doesn't need to do shift operation or type cast, so deal it in a separate logic.

Reply:
I know that, I meant this is true for all types. not necessarily for char. You could common up logic for both branches,

Reply:
Yes, we can do that.  The generated code may like the following:
`char _1;`
`char  _2 = ((char)((_1)<<0))`
That's correct, but the code is a little strange, and may takes more time in runtime? If that's OK, I think we can common up logic for both branches, what is your opinion?

Reply:
It should look like:

if lane == 1:
  return vec
else {
  stored as int and extracted with bit ops. 
}

Reply:
Do you mean store int8*2 as int16_t?

Reply:
I meant you have two "if (t.lanes() == 1)  return vec". You could just move this logic out, as it is true for all inputs. 

The other part is nice-to-have if you can naturally support char2/3/4./8. that will  help simplify the codegen logic. 

e.g. char2/3/4 can be stored as uint32_t. The only difference is only lower k bytes are useful.

Reply:
That's a good suggestion, thanks a lot, I will modify the code, and commit it later.

Reply:
@wpan11nv , I've realized the type with 'int',  and there is something wrong.
When the type is int8*2，the cuda code seems correct and also build correctly.
`
#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 610)

#include <sm_61_intrinsics.h>
#endif
extern "C" __global__ void default_function_kernel0(void* __restrict__ B, void* __restrict__ A) {
  int _1;
  {
    int _2 = (( int*)(( signed char*)A + (((((int)blockIdx.x) * 16) + (((int)threadIdx.x) * 2)))))[0];
    int _3 = (int)16843009;
    _1=((((char)(_2 >> 0))+((char)(_3 >> 0))) << 0);
    _1=_1 & ~(0x000000ff << 8) |((((char)(_2 >> 8))+((char)(_3 >> 8))) << 8);
  }
  (( int*)(( signed char*)B + (((((int)blockIdx.x) * 16) + (((int)threadIdx.x) * 2)))))[0] = _1;
}
`
But there is a runtime error when copy from gpu to cpu memory : Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading: misaligned address.  
Do you have any advice?

And if we use int32_t, when the type is int8*2, and if the size of the tensor is very huge, it will wastes lots of memory which is not nessary, and the runtime resource is more important. So I think based on the current code, just need a small modification that can support int8*3 by using char3 if needed support char3. What about your opinion?

Reply:
That is a good point I missed. We could not  use uint32_t as the stored type. for char2 or char3.  It is fine to keep it as it is. Thanks for looking into this!

Reply:
Thanks, I've moved the logic out, and modify code to support char3, add some unittest for char3.